test/reverseProxy.test.js

```diff
@@ -0,0 +1,123 @@
+var constants = require('../config/constants');
+var RdGlobalConfig = constants.RdGlobalConfig;
+var nock = require('nock');
+var RdHandler = require('../src/reverseProxy');
+var http = require('http');
+var assert = require('chai').assert;
+var testHelper = require('./testHelper');
+
+
+describe('RdHandler', function () {
+  context('Reverse Proxy Server', function () {
+    var originalScheme;
+    before(function (done) {
+      this.timeout = 5000;
+      testHelper.initializeDummyLoggers();
+      testHelper.initializeDummyHandlers();
+      originalScheme = RdGlobalConfig.SCHEME;
+      RdGlobalConfig.SCHEME = 'https';
+      
+      RdHandler.startReverseProxyServer(RdGlobalConfig.RD_HANDLER_REVERSE_PROXY_PORT, function (port) {
+        console.log('Test Network Utility Reverse Proxy Server Started on Port: ', port);
+        done();
+      });
+    });
+
+    after(function (done) {
+      this.timeout = 5000;
+      RdHandler.stopReverseProxyServer(function () {
+        done();
+      });
+      testHelper.deleteLoggers();
+      testHelper.deleteHandlers();
+      nock.cleanAll();
+      RdGlobalConfig.SCHEME.restore();
+      RdGlobalConfig.SCHEME = originalScheme;
+    });
+
+    it('Requests reverse proxy on behalf of the client and returns the response', function (done) {
+      this.timeout = 5000;
+      testHelper.nockGetRequest(constants.HUB_STATUS_URL, 'https', null, 200);
```

Comment:
The tests in `reverseProxy` are only for `https` it seems. Whereas in `proxy.test.js`, they are only for `http`. Shouldn't we have both in both cases?

Reply:
Since we have https support now for both...

Reply:
added tests

Reply:
great

Reply:
can be resolved.